parsers/wiki_parser.py

```diff
@@ -0,0 +1,291 @@
+    :return: GET parameters `params`.
+    """
+    params = FILE_API_PARAMS.copy()
+    if files:
+        titles = 'File:' + '|File:'.join(files)
+    else:
+        titles = ''
+    params['titles'] = titles
+    return params
+
+
+def is_hero_type(page):
+    """Method to check if page belongs to a hero, creep-hero(Warlock's Golem).
+
+    :param page: Page name as string.
+    :return: True if page belongs to hero else False
+    """
+    if '/Responses' in page:
+        return True
+    else:
+        return False
+
+
+def get_hero_name(hero_page):
+    """Method that parses hero name from its responses page.
+    Pages for heroes are in the form of `Hero name/Responses`. We need only the `Hero name` part for heroes.
+
+    :param hero_page: hero's responses page as string.
+    :return: Hero name as parsed
+    """
+    return hero_page.split('/')[0]
+
+
+def create_responses_text_and_link_list(responses_source):
+    """Method that for a given source of a hero's response page creates a list of tuple: (original_text, 
+    processed_text, link).
+    Steps involved:
+    * Use regex to find all lines containing mp3 files and responses.
+    * Process it to get original response text and file name.
+    * Create a list of files and get all the links for them by calling `links_for_files`.
+    * Process original text to get processed response.
+    * Add original response text, processed response text and file link to a list as a tuple.
+
+    :param responses_source: Mediawiki source
+    :return: list with tuples of (original_text, processed_text, link).
+    """
+    responses_list = []
+    file_and_text_list = []
+
+    response_regex = re.compile(RESPONSE_REGEX)
+    file_regex = re.compile(FILE_REGEX)
+
+    for response in response_regex.finditer(responses_source):
+        original_text = parse_response(response['text'])
+        if original_text == '':
+            pass
+
+        files_source = response['files']
+        for file in file_regex.finditer(files_source):
+            file_name = file['file'].replace('_', ' ').capitalize()
+            file_and_text_list.append([original_text, file_name])
+
+    files_list = [file for text, file in file_and_text_list]
+    file_and_link_dict = links_for_files(files_list)
+
+    for original_text, file in file_and_text_list:
+        processed_text = clean_response_text(original_text)
+        if processed_text != '':
+            try:
+                link = file_and_link_dict[file]
+                responses_list.append((original_text, processed_text, link))
+            except KeyError:
+                # Ignore files with no links to mp3 files. Happens to broken files and files undergoing migration.
+                pass
+
+    return responses_list
+
+
+def parse_response(text):
+    # Special cases
+    if '(broken file)' in text:
+        return ''
+    if 'versus (TI ' in text:
+        return ''
+    if 'Ceeeb' in text:
+        return ''
```

Comment:
"X or Y or Z" would be better in most cases as the returned default might change. In this case, the value is unlikely to change, yet in places using this function, the result is compared to '', so maybe the value should be saved as a constant and used everywhere as variable - easy to change, maintain and less bug prone. Right now chaning the value to None in one place, will cause a bug in another and it might not be obvious that all '' values should be changed together - with a constant it is obvious.

Reply:
Updated to return None and consolidated the if to a single liner in d266d930fbd4885d99aa0bd34300009ef2e3340c